src/tools/blast/components/results/BlastResultsButtons.tsx

```diff
@@ -1,11 +1,98 @@
 import React, { FC, lazy, useState, Suspense } from 'react';
-import { DownloadIcon, BasketIcon } from 'franklin-sites';
+import { useHistory } from 'react-router-dom';
+import { sleep } from 'timing-functions';
+import { DownloadIcon, BasketIcon, ReSubmitIcon } from 'franklin-sites';
+
 import SidePanel from '../../../../shared/components/layouts/SidePanel';
 
-const BlastResultsButtons: FC<{
+import { serverParametersToFormParameters } from '../../adapters/BlastParametersAdapter';
+
+import { LocationToPath, Location } from '../../../../app/config/urls';
+import uniProtKBApiUrls, {
+  getSuggesterUrl,
+} from '../../../../uniprotkb/config/apiUrls';
+
+import fetchData from '../../../../shared/utils/fetchData';
+
+import { PublicServerParameters } from '../../types/blastServerParameters';
+
+type ResubmitButtonProps = {
+  inputParamsData?: PublicServerParameters;
+};
+
+const ResubmitButton: FC<ResubmitButtonProps> = ({ inputParamsData }) => {
+  const history = useHistory();
+
+  const [disabled, setDisabled] = useState(false);
+
+  const handleClick = async () => {
+    if (!inputParamsData) {
+      return;
+    }
+
+    setDisabled(true);
+
+    const taxonMapping = new Map();
+    const taxonRequests = (inputParamsData.taxids || '')
+      .split(',')
+      .map((id) => {
+        const idCleaned = id.trim();
+        if (!idCleaned) {
+          return;
+        }
+        // eslint-disable-next-line consistent-return
+        return fetchData(
+          getSuggesterUrl(uniProtKBApiUrls.organismSuggester, idCleaned)
+        ).then((response) => {
+          const firstSuggestion = response?.data?.suggestions?.[0]?.value;
+          if (firstSuggestion) {
+            taxonMapping.set(
+              idCleaned,
+              `${firstSuggestion.replace(/ *\([^)]*\) */g, '')} [${idCleaned}]`
+            );
+          }
+        });
+      });
+
+    try {
+      // wait up to 2 seconds to get the information
+      await Promise.race([Promise.all(taxonRequests), sleep(2000)]);
+    } catch (_) {
+      /* */
```

Comment:
Should we just use the taxids in that case?

Reply:
I just thought it would look nicer to display the names in the same way that if you would have entered them through the taxon search box.
Also, I'm making sure that this is optionnal in the code, in the sense that if it fails to retrieve the information, or if takes too long, it is not blocking the flow of things and it would fallback to displaying the taxid as the label.

Reply:
Ok, hadn't seen that it was optional.